### PR TITLE
fix(snowflake): convert parameters to qmark style

### PR DESCRIPTION
**Pull request by hachichaud**
## Change Summary
The parameters value passed from the backend can be incompatible
with pyformat mapping (dict or array values).
I use convert to qmark method that is already used on other connectors
Another way would be to flatten parameters, but it could still leave
some incompatible values (arrays of dict).

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable

### CODE REVIEW

**Comment by raphaelvignes** [comment]
Thanks for this fix :+1:

**Review by raphaelvignes — APPROVED** [review]

File: tests/snowflake/test_snowflake.py [hunk 1]
```diff
@@ -36,7 +36,7 @@
     domain='test_domain',
     database='test_database',
     warehouse='test_warehouse',
-    query='test_query with %(foo)s and {{ pokemon }}',
+    query='test_query with %(foo)s and %(pokemon)s',
```
Comment (davinov) [inline tests/snowflake/test_snowflake.py:39]: `%(...)` is an old python syntax, I think we'd prefer to push Jinja' one in our connectors configs
Today, we're already replacing these automatically: https://github.com/ToucanToco/toucan-connectors/blob/ce8ed0622daf262abcec7f72bf0d90f6506c6962/toucan_connectors/common.py#L78

So my suggestion would be to use jinja {{ }} in your tests and documentation as well

**Review by davinov — DISMISSED** [review]
Nice :+1: 
A very small suggestion regarding delimiters used in examples:

File: tests/snowflake/test_snowflake.py (hunk 1 above)
Comment (hachichaud) [reply tests/snowflake/test_snowflake.py:39]: les syntaxes compatibles avec execute de snowflake sont

* pyformat (`%()s`)  => bug car on passe un dictionnaire parameters qui peut comporter des valeurs invalides.
* qmark => cest dans ce format qu'on convertit notre query pour bypasser le probleme du dictionnaire.

Par contre on ne peut pas utiliser jinja car ce n'est pas safe sur des requetes SQL
cf le code que tu pointes, qui est extrait de la methode no_sql_apply_parameters_to_query :

```
def nosql_apply_parameters_to_query(query, parameters, handle_errors=False):
    """
    WARNING : DO NOT USE THIS WITH VARIANTS OF SQL
    Instead use your client library parameter substitution method.
    https://www.owasp.org/index.php/Query_Parameterization_Cheat_Sheet
    """
```

on veut spécifiquement éviter le templating de cette manière.

**Comment by hachichaud** [comment]
[As a user i can use Snowflake live data queries on a chart or tile](https://trello.com/c/V84rCXYn/442-as-a-user-i-can-use-snowflake-live-data-queries-on-a-chart-or-tile)

**Review by davinov — APPROVED** [review]
We had a slack discussion about which style of param to use, let's open an issue/think more about it, but this PR is not the place for that.

File: tests/snowflake/test_snowflake.py (hunk 1 above)
Comment (hachichaud) [reply tests/snowflake/test_snowflake.py:39]: @davinov you already had an issue on that actually! https://github.com/ToucanToco/toucan-connectors/issues/303
^^

**Pull request merged**